StkAutomation/Matlab/STK Object Model/Radar_Scheduler_Deck_Access.m

```diff
@@ -0,0 +1,163 @@
+% %Written By: Alexander Ridgeway
+% %Date:       27 April 2016
+% 
+clear;
+clc;
+try
+    % Grab an existing instance of STK
+    uiapp = actxGetRunningServer('STK12.application');
+catch
+    % STK is not running, launch new instance
+    % Launch a new instance of STK and grab it
+    uiapp = actxserver('STK12.application');
+end
+
+%get the root from the personality
+root = uiapp.Personality2;
+% set visible to true (show STK GUI)
+uiapp.visible = 1;
+%%From the STK Object Root you can command every aspect of the STK GUI
+%close current scenario or open new one
+try
+    root.CloseScenario();
+    root.NewScenario('DeckAccess');
+catch
+    root.NewScenario('DeckAccess');
+end
+%get the scenario root, its of type IAgScenario 
+scen = root.CurrentScenario;
+%root.ExecuteCommand('SetUnits / Km Epsec');
+
+%Create the Radar Site (Delete if it exists)
+try
```

Comment:
I suggest replacing this with if statement check instead of try catch.

```matlab
if root.CurrentScenario.Children.Contains('eFacility', 'Radar')
    fac = root.GetObjectFromPath('/Facility/Radar');
    fac.Unload;
    fac = root.CurrentScenario.Children.New('eFacility','Radar');
else
    fac = root.CurrentScenario.Children.New('eFacility','Radar');
end
```